sycl/doc/extensions/proposed/SYCL_EXT_ONEAPI_GROUP_STATIC_ARRAY.asciidoc

```diff
@@ -0,0 +1,590 @@
+  void exclusive_scan_over_group(Group g, sycl::span<T, ItemsPerWorkItem> in_data, sycl::span<T, ItemsPerWorkItem> out_data, BinaryOperation binary_op, sycl::ext::oneapi::experimental::properties<Properties...> properties = {}); // (1)
+
+  template <typename Group, typename V, std::size_t ItemsPerWorkItem, typename T, typename BinaryOperation, typename... Properties>
+  void exclusive_scan_over_group(Group g, sycl::span<V, ItemsPerWorkItem> in_data, sycl::span<V, ItemsPerWorkItem> out_data, T init, BinaryOperation binary_op, sycl::ext::oneapi::experimental::properties<Properties...> properties = {}); // (2)
+
+}
+----
+
+_Preconditions_: `binary_op` must be an instance of a SYCL function object.
+
+1._Constraints_: Available only if `sycl::is_group_v<std::decay_t<Group>>` is true,
+`T` is a fundamental type, `BinaryOperation` is a SYCL function object type,
+`ItemsPerWorkItem` does not equal to `sycl::dynamic_extent`.
+
+_Mandates_: `binary_op(in_data[i], in_data[j])`, where `i, j < ItemsPerWorkItem`,
+must return a value of type `T`.
+
+_Effects_: Perform the exclusive scan of all values of `in_data` for all work-items in the group
+and the identity value of `binary_op`, using the operator `binary_op`. The scan is computed
+using a generalized noncommutative sum as defined in standard {cpp}. For multi-dimensional groups,
+the order of work-items in the group is determined by their linear id.
+Result is placed into `out_data` with the layout specified by `properties`.
+Default layout is one that is specified by the `close` property.
+`in_data` mustn't be changed by the algorithm.
+
+2._Constraints_: Available only if `sycl::is_group_v<std::decay_t<Group>>` is true,
+`V` and `T` are fundamental types, `BinaryOperation` is a SYCL function object type and
+`ItemsPerWorkItem` does not equal to `sycl::dynamic_extent`.
+
+_Mandates_: `binary_op(init, in_data[i])`, where `i < ItemsPerWorkItem`,
+must return a value of type `T`.
+
+_Effects_: Perform the exclusive scan of all values of `in_data` for all work-items in the group
+and an initial value specified by `init`, using the operator `binary_op`.
+The scan is computed using a generalized noncommutative sum as defined in standard {cpp}.
+For multi-dimensional groups, the order of work-items in the group is determined by their linear id.
+Result is placed into `out_data` with the layout specified by `properties`.
+Default layout is one that is specified by the `close` property.
+`in_data` mustn't be changed by the algorithm.
+
+[source,c++]
+----
+namespace sycl::ext::oneapi::experimental {
+
+  template <typename Group, typename T, std::size_t ItemsPerWorkItem, typename BinaryOperation, typename... Properties>
+  void inclusive_scan_over_group(Group g, sycl::span<T, ItemsPerWorkItem> in_data, sycl::span<T, ItemsPerWorkItem> out_data, BinaryOperation binary_op, sycl::ext::oneapi::experimental::properties properties = {}); // (1)
```

Comment:
Discussed the topic about properties with @rolandschulz. Decided to have following for sorting (the same for scans)
```cpp
  template <typename GroupHelper, typename T, std::size_t ItemsPerWorkItem, typename Properties = properties<>>
  void 
  sort_over_group(GroupHelper gh, sycl::span<T, ItemsPerWorkItem> values, Properties properties = {}); // (1)
```
mentioning `Available only if sycl::ext::oneapi::is_property_list_v<std::decay_t<Properties>> is true`

It will be better if we have any non-type parameters in the property list later.